tests/unittests/NodeSplittingTest.cpp

```diff
@@ -524,7 +776,6 @@ static void splitMaxPoolBasic(Function *F, Function *&optF,
 
   // Check node count.
   EXPECT_EQ(splitNodes.size(), totNumChunks);
-  EXPECT_EQ(countNodeKind(F, Kinded::Kind::SliceNodeKind), totNumChunks);
```

Comment:
Quick question, why don't we need to check for slices here in these tests anymore?

Reply:
With this PR Slice nodes are no longer inserted when not necessary. For example when splitting a Convolution along H dimension, all the Convolution parts will reuse same filter (entirely) so the filter will not be sliced anymore for each part. This modification just makes a little less mess when splitting nodes. In the tests this means the number of Slice nodes is no longer that predictable therefore I erased the checks.